tools/functional-tester/etcd-runner/command/election_command.go

```diff
@@ -65,18 +71,30 @@ func runElectionFunc(cmd *cobra.Command, args []string) {
 				break
 			}
 		}
-		e := concurrency.NewElection(s, "electors")
 
-		rcs[i].acquire = func() error {
-			<-releasec
+		e := concurrency.NewElection(s, election)
+		rcs[i].acquire = func() (err error) {
 			ctx, cancel := context.WithCancel(context.Background())
+			donec := make(chan struct{})
 			go func() {
-				if ol, ok := <-e.Observe(ctx); ok {
+				defer close(donec)
+				ol, ok := <-e.Observe(ctx)
```

Comment:
OK, but the loop I suggested creates a new observe channel on each iteration. It shouldn't get stuck on the compaction revision because observe channels initialize using the current revision.